Create integration test for dynamic nodes not powering down after cluster update

jdeamicis opened the pull request:
### Description of changes
* Add integration test for dynamic nodes not powering down after cluster update

### Tests
* Ran implemented integration test with and without sleep time between restart of slurmctld and scontrol reconfigure

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [x] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

jdeamicis committed:
Create integration test for dynamic nodes not powering down after cluster update

Signed-off-by: Jacopo De Amicis <jdamicis@amazon.it>

demartinofra commented on tests/integration-tests/tests/update/test_update.py:
```diff
@@ -653,3 +655,74 @@ def _test_update_queue_strategy_with_running_job(
+
+
+@pytest.mark.usefixtures("region", "os", "instance", "scheduler")
+def test_update_slurm_reconfigure_race_condition(
```
since this is Slurm specific bug I'd keep this test under the test_slurm.py

jdeamicis replied:
Addressed in the latest commit.

demartinofra commented on tests/integration-tests/tests/update/test_update.py:
```diff
@@ -653,3 +655,74 @@ def _test_update_queue_strategy_with_running_job(
+    """
+    Test race condition between restart of slurmctld and scontrol reconfigure.
+
+    In Slurm 21.08 it looks like cloud nodes may not get powered-down after their
```
worth adding a link to the ticket

jdeamicis replied:
Addressed in the latest commit.

demartinofra commented on tests/integration-tests/tests/update/test_update.py:
```diff
@@ -653,3 +655,74 @@ def _test_update_queue_strategy_with_running_job(
+
+    assert_compute_node_states(slurm_commands, compute_nodes=None, expected_states=["idle~"])
+
+    for iter in range(1, 6):
```
nit: assign the retries to a constant/variable

jdeamicis replied:
I'm not sure I understand what you mean exactly here.

jdeamicis replied:
OK, we had a separate chat about this. Addressed in the latest commit.

jdeamicis committed:
Address comments in PR

Move test to test_slurm.py

Add link to SchedMD bug in test description

Add variable for max_retries

Signed-off-by: Jacopo De Amicis <jdamicis@amazon.it>
